src/msg/async/dpdk/DPDKStack.cc

```diff
@@ -250,8 +250,14 @@ void DPDKStack::spawn_worker(unsigned i, std::function<void ()> &&func)
   // if dpdk::eal::init already called by NVMEDevice, we will select 1..n
   // cores
   assert(rte_lcore_count() >= i + 1);
+  unsigned core_id;
+  RTE_LCORE_FOREACH_SLAVE(core_id) {
+    if (i-- == 0) {
+      break;
+    }
+  }
```

Comment:
what's the meaning?

Reply:
I think the idea of  #20638 would be better

Reply:
@yuyuyu101 it's to get the core id from the its index. for example, if i is 0, the first non master core id is used.

Reply:
@shangfufei, i'd be happy to learn the rationale behind your assertion.